src/coreclr/System.Private.CoreLib/src/System/Threading/ThreadPool.CoreCLR.Core.cs

```diff
@@ -0,0 +1,192 @@
+
+namespace System.Threading
+{
+    internal sealed partial class CompleteWaitThreadPoolWorkItem : IThreadPoolWorkItem
+    {
+        void IThreadPoolWorkItem.Execute() => CompleteWait();
+
+        // Entry point from unmanaged code
+        private void CompleteWait()
+        {
+            PortableThreadPool.CompleteWait(_registeredWaitHandle, _timedOut);
+        }
+    }
+
+    public static partial class ThreadPool
+    {
+        private static bool EnsureConfigInitializedCore()
+        {
+            return s_initialized;
+        }
+
+        private static readonly bool s_initialized = InitializeConfig();
+
+        // Indicates whether the thread pool should yield the thread from the dispatch loop to the runtime periodically so that
+        // the runtime may use the thread for processing other work
+        internal static bool YieldFromDispatchLoop => false;
+
+        private static readonly bool IsWorkerTrackingEnabledInConfig = GetEnableWorkerTracking();
+
+        private static unsafe bool InitializeConfig()
+        {
+            int configVariableIndex = 1;
+            while (true)
+            {
+                int nextConfigVariableIndex =
+                    GetNextConfigUInt32Value(
+                        configVariableIndex,
+                        out uint configValue,
+                        out bool isBoolean,
+                        out char* appContextConfigNameUnsafe);
+                if (nextConfigVariableIndex < 0)
+                {
+                    break;
+                }
+
+                Debug.Assert(nextConfigVariableIndex > configVariableIndex);
+                configVariableIndex = nextConfigVariableIndex;
+
+                Debug.Assert(appContextConfigNameUnsafe != null);
+
+                var appContextConfigName = new string(appContextConfigNameUnsafe);
+                if (isBoolean)
+                {
+                    AppContext.SetSwitch(appContextConfigName, configValue != 0);
+                }
+                else
+                {
+                    AppContext.SetData(appContextConfigName, configValue);
+                }
+            }
+
+            return true;
+        }
+
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        private static extern unsafe int GetNextConfigUInt32Value(
+            int configVariableIndex,
+            out uint configValue,
+            out bool isBoolean,
+            out char* appContextConfigName);
+
+        private static bool GetEnableWorkerTracking() =>
+            AppContextConfigHelper.GetBooleanConfig("System.Threading.ThreadPool.EnableWorkerTracking", false);
+
+        /// <summary>
+        /// Gets the number of thread pool threads that currently exist.
+        /// </summary>
+        /// <remarks>
+        /// For a thread pool implementation that may have different types of threads, the count includes all types.
+        /// </remarks>
+        public static int ThreadCount
```

Comment:
Everything starting with ThreadCount till end of this file should be moved under libraries